server/services/admin.ts

```diff
@@ -210,18 +217,105 @@ export = ({ strapi }: StrapiContext): IServiceAdmin => ({
     };
   },
 
+  //Find all reports
+  async findReports(
+    this: IServiceAdmin,
+    {...query}: AdminFindAllProps,
+  ): Promise<ToBeFixed> {
+    const {
+      _q,
+      orderBy,
+      pageSize = 10,
+      page = 1,
+      filters,
+    }: AdminFindAllQueryParamsParsed = parseParams<AdminFindAllQueryParamsParsed>(
+      query,
+    );
+
+    const defaultWhere = {
+      $or: [{resolved: false}, {resolved: true}],
+    };
+
+    const defaultAuthorUserPopulate = this.getDefaultAuthorPopulate();
+
+    let params: StrapiDBQueryArgs<CommentModelKeys> = {
+      where: !isEmpty(filters)
+        ? {
+            ...defaultWhere,
+            ...filters,
+          }
+        : {...defaultWhere},
+      offset: (page - 1) * pageSize,
+      limit: pageSize,
+      orderBy: orderBy || [{resolved: 'asc'}, {createdAt: 'desc'}],
+    };
+    if (_q) {
+      params = {
+        ...params,
+        where: {
+          ...params.where,
+          content: {
+            $contains: _q,
+          },
+        },
+      };
+    }
+
+    const entities = await strapi.db
+      .query<Comment>(getModelUid('comment-report'))
+      .findMany({
+        ...params,
+        populate: {
+          related: true,
+        },
```

Comment:
```suggestion
        populate: [ "related" ],
```
To be consistent with existing code.